src/content/en/fundamentals/native-hardware/capturing-images/index.md

```diff
@@ -2,210 +2,343 @@ project_path: /web/_project.yaml
 
 ## Access the camera interactively
+Now that we've covered our bases, it's time to progressively enhance!
 
 Modern browsers can get direct access to cameras, allowing us to build
 experiences that are fully integrated with the web page, so the user need never
 leave the browser.
 
-Warning: Direct access to the camera is a powerful feature, it requires consent 
-from the user and your site needs to be on a secure origin (HTTPS).
-
 ### Acquire access to the camera
 
-We can directly access a camera and microphone by using an API in the WebRTC 
-specification called `getUserMedia()`. This will prompt the user for 
+We can directly access a camera and microphone by using an API in the WebRTC
+specification called `getUserMedia()`. This will prompt the user for
 access to their connected microphones and cameras.
 
+Support for `getUserMedia` is pretty good, but it isn't yet everywhere. In particular, it is not
+available in Safari 10 or lower, which at the time of writing is still the latest stable version.
+However, [Apple have announced](https://webkit.org/blog/7726/announcing-webrtc-and-media-capture/)
+that it will be available in Safari 11.
+
+It's very simple to detect support, however.
+
+    const supported = 'mediaDevices' in navigator;
+
+Warning: Direct access to the camera is a powerful feature. It requires consent
+from the user, and your site MUST be on a secure origin (HTTPS).
+
+When you call `getUserMedia`, you need to pass in an object that describes what kind of media you
+want. These choices are called constraints.
+
+There are a several possible constraints, covering things like whether you prefer a front- or
+rear-facing camera, whether you want audio, and your preferred resolution for the stream.
+
+To get data from the camera, however, we need just one constraint, and that is `video: true`.
+
 If successful the API will return a `MediaStream` that contains data from
 the camera, and we can then either attach it to a `<video>` element and play it
 to show a real time preview, or attach it to a `<canvas>` to get a
 snapshot.
 
-To get data from the camera we just set `video: true` in the constraints 
-object that is passed to the `getUserMedia()` API.
-
     <video id="player" controls autoplay></video>
-    <script>  
-      var player = document.getElementById('player');
+    <script>
+      const player = document.getElementById('player');
 
-      var handleSuccess = function(stream) {
-        player.srcObject = stream;
+      const constraints = {
+        video: true,
       };
 
-      navigator.mediaDevices.getUserMedia({video: true})
-          .then(handleSuccess);
+      navigator.mediaDevices.getUserMedia(constraints)
+        .then((stream) => {
+          player.srcObject = stream;
+        });
     </script>
 
 By itself, this isn't that useful. All we can do is take the video data and play
-it back.
+it back. If we want to get an image, we have to do a little extra work.
```

Comment:
You should be somewhat consistent about using I, you, and we in an article. You've been fairly consistent about using 'you' up until now, so go with that. (Also, the DevSite style guide discourages using 'we'.)